public/app/features/dashboard/state/runRequest.ts

```diff
@@ -34,14 +34,14 @@ export function processResponsePacket(packet: DataQueryResponse, state: RunningQ
   packets[packet.key || 'A'] = packet;
 
   // Update the time range
-  let timeRange = request.range;
-  if (isString(timeRange.raw.from)) {
-    timeRange = {
-      from: dateMath.parse(timeRange.raw.from, false),
-      to: dateMath.parse(timeRange.raw.to, true),
-      raw: timeRange.raw,
-    };
-  }
+  const range = request.range;
+  const timeRange = isString(range.raw.from)
```

Comment:
I think it's clearer that the timeRange has changed if we return undefined or an object. We could also always return a timeRange and that way it would be easier for the consumers?